tests/unit/test_state.py

```diff
@@ -610,3 +627,149 @@ def test_builder_app_channel_from_charm_error():
         state.BuilderAppChannel.from_charm(charm=charm)
 
     assert "invalid" in str(exc.getrepr())
+
+
+@pytest.mark.parametrize(
+    "exception, expected_error",
+    [
+        pytest.param(ops.SecretNotFoundError, "Secret label not found:"),
+        pytest.param(ops.ModelError, "Please grant the charm read access to the secret."),
+    ],
+)
+@pytest.mark.usefixtures("patch_juju_version_33")
+def test__parse_script_secrets_invalid_secret(exception: Exception, expected_error: str):
+    """
+    arrange: given a mocked model get_secret method that raises a given error.
+    act: when _parse_script_secrets is called.
+    assert: InvalidSecretError is raised.
+    """
+    mock_charm = MagicMock()
+    mock_charm.config = {state.SCRIPT_SECRET_ID_CONFIG_NAME: "secret-label"}
+    mock_charm.model = MagicMock()
+    mock_charm.model.get_secret = MagicMock(side_effect=exception)
+
+    with pytest.raises(state.InvalidSecretError) as exc:
+        state._parse_script_secrets(charm=mock_charm)
+
+    assert expected_error in str(exc)
+
+
+@pytest.mark.usefixtures("patch_juju_version_33")
+def test__parse_script_secrets_secret_and_config_set():
+    """
+    arrange: given a config option where both secret and the config is set.
+    act: when _parse_script_secrets is called.
+    assert: InvalidSecretError is raised.
+    """
+    mock_charm = MagicMock()
+    mock_charm.config = {
+        state.SCRIPT_SECRET_ID_CONFIG_NAME: "secret:test-secret-id",
+        state.SCRIPT_SECRET_CONFIG_NAME: "test-secret",
+    }
+
+    with pytest.raises(state.InvalidSecretError) as exc:
+        state._parse_script_secrets(charm=mock_charm)
+
+    assert "Both script-secret and script-secret-id configuration option set." in str(exc)
+
+
+def test__parse_script_secrets_secret_unsupported(monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: given a Juju version < 3.3 and secret config option set.
+    act: when _parse_script_secrets is called.
+    assert: InvalidSecretError is raised.
+    """
+    monkeypatch.setattr(
+        ops.JujuVersion, "from_environ", MagicMock(return_value=ops.JujuVersion("3.2"))
+    )
+    mock_charm = MagicMock()
+    mock_charm.config = {
+        state.SCRIPT_SECRET_ID_CONFIG_NAME: "secret:test-secret-id",
+    }
+
+    with pytest.raises(state.InvalidSecretError) as exc:
+        state._parse_script_secrets(charm=mock_charm)
+
+    assert "Secrets are not supported in Juju version" in str(exc)
+
+
+def test__parse_script_prefer_secrets(monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: given a Juju version > 3.3 and secret config option set.
+    act: when _parse_script_secrets is called.
+    assert: InvalidSecretError is raised.
+    """
+    monkeypatch.setattr(
+        ops.JujuVersion, "from_environ", MagicMock(return_value=ops.JujuVersion("3.5"))
+    )
```

Comment:
couldn't we use the fixture `patch_juju_version_33` (which is returning juju 3.5) instead?